### PR TITLE
Add k8s token fetch

### PR DESCRIPTION
**Pull request by Hosweay**
A small refactor and then adding Kubernetes Vault support

Testing done
Mesos
- Marathon
- Chronos

Kubernetes

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -21,32 +24,120 @@ type vaultExecConfig struct {
...
 
-// Do a case insensitive search for MESOS_TASK_ID
-func getTaskId(envVars []string) (string, error) {
-	for _, envVar := range envVars {
-		res := strings.SplitN(envVar, "=", 2)
-		if strings.ToUpper(res[0]) == "MESOS_TASK_ID" {
-			return res[1], nil
-		}
+func determineScheduler() (string, error) {
+	if os.Getenv("MESOS_TASK_ID") != "" {
```
Comment (TheDahv) [inline main.go]: Wait, do you mean to return the "mesos" default when the environment variable is not empty?

Comment (TheDahv) [reply main.go]: Oh, maybe I need to a do a better job understanding what this does. Is the idea here saying that you need either one of these set, and then that determines what scheduler it looks for?

**Comment by funnylookinhat** [comment]
@Hosweay I know you already merged it - so this comment is sort of a throwaway I guess...

I think it'd be a good idea to document the functions in this app and even add some more doc throughout them.  I know we all understand what it's doing (it's pretty obvious, etc.) but whoever comes after us is going to be using this source code as documentation most likely - couldn't hurt to be overly explicit even with these sorts of things.

File: main.go (hunk 1 above)
Comment (Hosweay) [reply main.go]: Yeah. I am just trying to determine by the environment variables set which container scheduler is being used. To use vault-gatekeeper the MESOS_TASK_ID var must be set and Mesos always sets it. The other env is just something that k8s sets that I can check if the container is running in k8s.

**Pull request merged**